hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RegionScannerImpl.java

```diff
@@ -768,7 +768,7 @@ public synchronized boolean reseek(byte[] row) throws IOException {
   }
 
   @Override
-  public void shipped() throws IOException {
+  public synchronized void shipped() throws IOException {
```

Comment:
Why this change?

Reply:
Because in the StoreScanner, the` prevCell` will be copied to heap before ship(), but it may be null even after the close() of the store scanner. As a result, when there are concurrent close() and ship() of a scanner, and heap cleared before ship, the `prevCell` block will be released before copy the cell.
In RPC handlers, suppose a situation, two scan() RPCs use the same scanner, and one got partial results, one found that no more results, then in the RPC callback, there is no guarantee for the close callback happens before the ship callback, then when ship() copy `prevCell`, the close may have already released previous blocks...

Reply:
Is it possible for two scan() RPCs use the same scanner? Just want to make sure this is necessary...

Reply:
Hi, @saintstack , @Apache9 , actually we didn't find any clue of the concurrently ship() and close() in RPC handlers, but there are some when compaction. 
@Apache9  do you mean the scanner lease may help in this problem? One RPC handler gets a cached scanner and removes the lease, till encounters ex or ship() completes in shipped-callback will add the scanner lease back. The other RPC handlers should wait the lease exits, or else will throw ex.

Reply:
We can add some protect codes before clone the previous cell and ship the previous blocks in the store scanner. I found the warn log occurs in compaction.
![image](https://user-images.githubusercontent.com/10123703/129008677-6894b638-bf38-4f6d-b033-ebeaefd9a0c5.png)

Reply:
So, you want to just land the bucketEntry.isRpcRef check for now @sunhelly ? I'm good with that.

Reply:
Yes, @saintstack , we can fix the key issue here first, and then optimize other parts if necessary in the future. Thanks.

Reply:
Agree. Let's land the isRpcRef thing first, And then let's discuss how to correct do synchronization.